src/main/scala/org/neo4j/spark/LazySessionIterator.scala

```diff
@@ -0,0 +1,32 @@
+package org.neo4j.spark
+
+import org.neo4j.driver.Driver
+import org.neo4j.driver.Session
+
+class LazySessionIterator[T](iterator: Iterator[T], driver: Driver, session: Session) extends Iterator[T] {
```

Comment:
I'm not suer this really works ... can this leak out of scope of the worker job so that the driver/session might not be closed (if no one calls close here) there is no guarantee or?